src/Simulations/callback.jl

```diff
@@ -51,3 +51,51 @@ Callback(wta::WindowedTimeAverage, schedule; kw...) =
     throw(ArgumentError("Schedule must be inferred from WindowedTimeAverage. 
                         Use Callback(windowed_time_average)"))
 
+struct GenericName end
+
+function unique_callback_name(name, existing_names)
+    if name ∈ existing_names
+        return Symbol(:another_, name)
+    else
+        return name
+    end
+end
+
+function unique_callback_name(::GenericName, existing_names)
+    prefix = :callback # yeah, that's generic
+
+    # Find a unique one
+    n = 1
+    while Symbol(prefix, n) ∈ existing_names
+        n += 1
+    end
+
+    return Symbol(prefix, n)
+end
+
+"""
+    add_callback!(simulation, callback::Callback; name = GenericName())
+                  
+    add_callback!(simulation, func; schedule=IterationInterval(1), name = GenericName())
+
+Add `Callback(func, schedule)` to `simulation.callbacks` under `name`. The default
+`GenericName()` generates a name of the form `:callbackN`, where `N`
+is big enough for the name to be unique.
+
+If `name` is supplied, it may be modified if `simulation.callbacks[name]`
```

Comment:
is `name` expected as string or symbol? should we clarify?

Reply:
I understand that's a symbol...

Reply:
Yes, let's clarify that